### PR TITLE
Created few tests. Moved method getRouteKey() to new abstract class Abs…

### PR DESCRIPTION
**Pull request by Alexey007gold**
…tractByCommandRouterHandler.

### CODE REVIEW

File: basis/common-router/src/main/java/com/alex_koveckiy/common/router/api/AbstractByCommandRouterHandler.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package com.alex_koveckiy.common.router.api;
+
+import com.alex_koveckiy.common.protocol.Request;
+
+/**
+ * Created by alex on 02.03.17.
+ */
+public abstract class AbstractByCommandRouterHandler<T extends Handler> extends AbstractRouterHandler<T> {
```
Comment (Dominys) [inline basis/common-router/src/main/java/com/alex_koveckiy/common/router/api/AbstractByCommandRouterHandler.java:8]: @Alexey007gold this is not abstract

File: basis/common-router/src/test/java/com/alex_koveckiy/common/router/AbstractByCommandRouterHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+/**
+ * Created by alex on 02.03.17.
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class AbstractByCommandRouterHandlerTest {
+
+    @InjectMocks
+    private AbstractByCommandRouterHandler<Handler> handler =
+            Mockito.mock(AbstractByCommandRouterHandler.class, Mockito.CALLS_REAL_METHODS);
```
Comment (Dominys) [inline basis/common-router/src/test/java/com/alex_koveckiy/common/router/AbstractByCommandRouterHandlerTest.java:24]: @Alexey007gold No need to use mock if you use @InjectMocks. Or dont use InjectMocks - you dont have any mocks to inject.

File: basis/common-router/src/test/java/com/alex_koveckiy/common/router/AbstractRouterHandlerTest.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+/**
+ * Created by alex on 02.03.17.
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class AbstractRouterHandlerTest {
+
+    @InjectMocks
+    private AbstractRouterHandler<Handler> handler =
+            Mockito.mock(AbstractRouterHandler.class, Mockito.CALLS_REAL_METHODS);
```
Comment (Dominys) [inline basis/common-router/src/test/java/com/alex_koveckiy/common/router/AbstractRouterHandlerTest.java:27]: @Alexey007gold Same

File: basis/common-router/src/test/java/com/alex_koveckiy/common/router/AbstractRouterHandlerTest.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+    @Spy
+    private HandlerFactory handlerFactory;
```
Comment (Dominys) [inline basis/common-router/src/test/java/com/alex_koveckiy/common/router/AbstractRouterHandlerTest.java:30]: @Alexey007gold Why you need spy ?

File: basis/common-router/src/test/java/com/alex_koveckiy/common/router/FirstRouterTest.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+/**
+ * Created by alex on 02.03.17.
+ */
+
+@RunWith(MockitoJUnitRunner.class)
+public class FirstRouterTest {
+
+    @InjectMocks
+    private FirstRouter firstRouter;
```
Comment (Dominys) [inline basis/common-router/src/test/java/com/alex_koveckiy/common/router/FirstRouterTest.java:22]: @Alexey007gold If you use inject - than inject something!

File: basis/common-router/src/test/java/com/alex_koveckiy/common/router/FirstRouterTest.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertThat;
+
+/**
+ * Created by alex on 02.03.17.
+ */
+
+@RunWith(MockitoJUnitRunner.class)
+public class FirstRouterTest {
```
Comment (Dominys) [inline basis/common-router/src/test/java/com/alex_koveckiy/common/router/FirstRouterTest.java:19]: @Alexey007gold Where is tests for handle method ?

**Comment by Dominys** [comment]
@Alexey007gold If you use InjectMocks annotation - than you have to inject something!

File: basis/common-router/src/main/java/com/alex_koveckiy/common/router/api/AbstractByCommandRouterHandler.java (hunk 1 above)
Comment (Alexey007gold) [reply basis/common-router/src/main/java/com/alex_koveckiy/common/router/api/AbstractByCommandRouterHandler.java:8]: Why? In what sense?

**Pull request closed without merging**